pkg/auth/providers/common/provider_util.go

```diff
@@ -24,3 +30,39 @@ func EscapeUUID(s string) string {
 	}
 	return buffer.String()
 }
+
+// Decode will decode to the output structure by creating a custom decoder
+// that uses the stringToK8sTimeHookFunc to handle the metav1.Time field properly.
+func Decode(input, output any) error {
```

Comment:
I think this function needs a unit test.

Reply:
Good idea.  Test added.